ui/src/app/pages/workspace/workspace-card.tsx

```diff
@@ -153,17 +155,29 @@ export const WorkspaceCard = fp.flow(withNavigation)(
       ]);
     }
 
-    onClick() {
+    // If enableResearchReviewPrompt is true and workspace needs Review:
+    // show the research purpose review modal and return false, to stop further navigation
+    // Else:
+    // Tracks the Analytics Event, Navigate to Featured workspace or Navigate to User workspace
+    // and return true to continue navigate to workspace detail page
+    trackWorkspaceNavigation(): boolean {
```

Comment:
looks like the logic here is now exactly the same as it was before.  Is that correct?  The previous version is less complex than this one, so I'd prefer to revert the change.

Reply:
How is that less complex? This is just separating the navigation vs tracking, Also i need to work on it a little more as per our conversation on friday

Reply:
Sorry!  I forgot that this was in-progress... LMK when you're ready for review or if you want to chat more.